src/cnd/swaps_payload.ts

```diff
@@ -79,3 +80,227 @@ export type HalightLightningBitcoinHerc20EthereumErc20RequestBody = CoreRequestB
+     */
+    // Still not really happy here re the best way to move forward.
+    asset: Asset;
+
+    // expiry: number; // No expiry needed, the refund can be handled via the actions
+  };
+}
+
+/**
+ * The detailed description of the ledger state.
+ */
+export interface LedgerState extends EmbeddedRepresentationSubEntity {
+  /**
+   * The relation of this ledger state to the parent object (*SwapProperties).
+   */
+  rel: ["alpha" | "beta"];
+  /**
+   * Human readable title.
+   */
+  title: "State of the Alpha Ledger" | "State of the Beta Ledger";
+  /**
+   * Class of this subentity to facilitate parsing.
+   */
+  class: ["state"];
+  // htlc_location?: any; // Not included because the user can search the deploy or fund tx in an explorer to find it.
+
+  properties: {
+    /**
+     * Blockchain Transactions
+     */
+    transactions: StepTransaction[];
+    /**
+     * The status of the escrow (HTLC or Invoice).
+     */
+    status: EscrowStatus;
+  };
+}
+
+/**
+ * The blockchain transactions related to a given step.
+ */
+export type StepTransaction = {
+  /**
+   * List the transaction related to the swap by step, if applicable.
+   *
+   * Key is the step, value is the transaction id
+   */
+  [k in Step]: string;
+};
+
+/**
+ * The status of the escrow (htlc, lightning invoice, etc) on the ledger.
+ */
+export enum EscrowStatus {
+  /**
+   * The escrow does not exist yet.
+   */
+  None = "NONE",
+  /**
+   * The escrow has been initialized.
+   *
+   * Initialization is a step that does not endure any cost to the user.
+   */
+  Initialized = "INITIALIZED",
+  /**
+   * The escrow has been deployed.
+   *
+   * Deployment is a step that endures some, relatively small, cost to the user due to computation needed on the blockchain.
+   */
+  Deployed = "DEPLOYED",
+  /**
+   * The escrow has been funded.
+   *
+   * Funding is a step where all the assets to be sold are sent and locked in the escrow.
+   */
+  Funded = "FUNDED",
+  /**
+   * The assets have been redeemed from the escrow.
+   *
+   * Redemption is a step where all the assets to be acquired are received from the escrow.
+   */
+  Redeemed = "REDEEMED",
+  /**
+   * The assets have been refunded from the escrow.
+   *
+   * Refunding is a step where all the assets to be sold are received back from the escrow, meaning the swap has been aborted.
+   */
+  Refunded = "REFUNDED",
+  /**
+   * An incorrect amount of assets have been sent to the escrow.
+   *
+   * To protect the user, if an incorrect amount of asset have been sent to the escrow, cnd will not propose redemption
+   * as an option and only the refund actions will be available down the line.
+   */
+  IncorrectlyFunded = "INCORRECTLY_FUNDED"
+}
+
+/**
+ * The identities related to a given party.
+ */
+export interface Identities extends EmbeddedRepresentationSubEntity {
+  /**
+   * The relation of these identities to the parent object (*SwapProperties).
+   */
+  rel: ["self" | "peer"];
```

Comment:
self is actually a registered relation and means something very different :)

Reply:
What do you think would make more sense?
- local
- own
- ..?

Reply:
identities removed